### PR TITLE
Draw basic shapes - V1

### PR DESCRIPTION
**Pull request by arshithsuresh**

### CODE REVIEW

File: src/app/canvas/canvas.component.ts [hunk 1]
```diff
@@ -7,25 +8,24 @@ import { Component, OnInit, ViewChild,ElementRef, AfterViewInit } from '@angular
...
     else
     {
-      console.log("Canvas Null!")
+      console.log("Canvas Null!");      
     }
   }
 
   ngOnInit(): void {
     
   }
```
Comment (isham-mohamed) [inline src/app/canvas/canvas.component.ts]: initialise fabric here

File: src/app/core/canvas/canvas-core.service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+})
+export class CanvasCoreService {
+  
+  objects:Map<string,IDrawableObject>= new Map();
+  canvasLoaded:EventEmitter<HTMLCanvasElement> = new EventEmitter();
+  canvas!:Canvas;
+  constructor() {
+    this.canvasLoaded.subscribe((newCanvas)=>{
+      this.canvas = new fabric.Canvas(newCanvas); 
+
```
Comment (isham-mohamed) [inline src/app/core/canvas/canvas-core.service.ts]: move this to canvas compoenent

File: src/app/side-bar-pallete/collections/create-shape.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { IDrawableObject } from "src/app/core/objects/object";
+
+ export interface CreateShape {
+    
+    myShape? : IDrawableObject;    
+    createShape():void;
+}
```
Comment (isham-mohamed) [inline src/app/side-bar-pallete/collections/create-shape.ts:12]: delete

File: src/app/canvas/canvas.component.ts (hunk 1 above)
Comment (arshithsuresh) [reply src/app/canvas/canvas.component.ts]: Initialized Fabric canvas in canvas component.

File: src/app/core/canvas/canvas-core.service.ts (hunk 2 above)
Comment (arshithsuresh) [reply src/app/core/canvas/canvas-core.service.ts]: Moved to canvas component

File: src/app/canvas/canvas.component.ts [hunk 4]
```diff
@@ -7,25 +10,35 @@ import { Component, OnInit, ViewChild,ElementRef, AfterViewInit } from '@angular
...
+  }
 
   ngAfterViewInit(): void {
-    if(this.mcanvas!=null)
+    if(this._mCanvas!=null)
     {
-      this.context = this.mcanvas.nativeElement.getContext('2d');      
+      this._mCanvas.nativeElement.width  = this._mCanvas.nativeElement.offsetWidth;
+      this._mCanvas.nativeElement.height = this._mCanvas.nativeElement.offsetHeight;
+      this.canvas = new fabric.Canvas(this._mCanvas.nativeElement);
```
Comment (isham-mohamed) [inline src/app/canvas/canvas.component.ts]: You can use either this or just pass the canvas id to initialise fabric

File: src/app/core/canvas/canvas-core.service.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class CanvasCoreService {
+  
+  objects:Map<string,IDrawableObject>= new Map();  
+  canvas!:Canvas;
+  
+  shapeCreated : EventEmitter<IDrawableObject> = new EventEmitter();
```
Comment (isham-mohamed) [inline src/app/core/canvas/canvas-core.service.ts]: Event emitters are usually used with @Output Decorators, Use Subject instead

Comment (isham-mohamed) [inline src/app/core/canvas/canvas-core.service.ts]: Use Subject

**Review by isham-mohamed — CHANGES_REQUESTED** [review]

File: src/app/core/canvas/canvas-core.service.ts (hunk 5 above)
Comment (arshithsuresh) [reply src/app/core/canvas/canvas-core.service.ts]: Changed to Subject.

File: src/app/canvas/canvas.component.ts (hunk 4 above)
Comment (arshithsuresh) [reply src/app/canvas/canvas.component.ts]: Tried with canvas id.

**Review by isham-mohamed — APPROVED** [review]

**Pull request merged**